version: '3.8'

services:
  web:
    build: .
    ports:
      - "8000:8000"
    environment:
      - SQLALCHEMY_DATABASE_URL=${SQLALCHEMY_DATABASE_URL}
    depends_on:
      db:
        condition: service_healthy
    networks:
      - pulse-network

  db:
    image: mysql:8.0
    environment:
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
      MYSQL_DATABASE: ${MYSQL_DATABASE}
      MYSQL_USER: user
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
    volumes:
      - mysql_data:/var/lib/mysql
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      timeout: 20s
      retries: 10
    networks:
      - pulse-network

volumes:
  mysql_data:

networks:
  pulse-network:
    driver: bridge
